hudi-client/hudi-client-common/src/main/java/org/apache/hudi/table/HoodieTimelineArchiveLog.java

```diff
@@ -321,10 +321,19 @@ public void archive(HoodieEngineContext context, List<HoodieInstant> instants) t
       List<IndexedRecord> records = new ArrayList<>();
       for (HoodieInstant hoodieInstant : instants) {
         try {
-          deleteAnyLeftOverMarkers(context, hoodieInstant);
-          records.add(convertToAvroRecord(hoodieInstant));
-          if (records.size() >= this.config.getCommitArchivalBatchSize()) {
-            writeToFile(wrapperSchema, records);
+          if (table.getActiveTimeline().isEmpty(hoodieInstant)
+                  && (
+                          hoodieInstant.getAction().equals(HoodieTimeline.CLEAN_ACTION)
+                          || (hoodieInstant.getAction().equals(HoodieTimeline.ROLLBACK_ACTION) && hoodieInstant.isCompleted())
+                     )
```

Comment:
We better add a uniform util do decide what kind of instant file(action type and state) should be non-empty.

Reply:
None of them should really be empty. is this happening on HDFS or cloud storage?

Reply:
HDFS